shell/renderer/api/atom_api_context_bridge.cc

```diff
@@ -0,0 +1,408 @@
+      key_str = base::NumberToString(key_int);
+    }
+    v8::Local<v8::Value> value;
+    if (!api.Get(key_str, &value))
+      continue;
+
+    if (value->IsFunction()) {
+      auto func = v8::Local<v8::Function>::Cast(value);
+      v8::Global<v8::Function> global_func(api.isolate(), func);
+
+      size_t func_id = store->take_id();
+      store->functions()[func_id] =
+          std::make_tuple(std::move(global_func),
+                          v8::Global<v8::Context>(source_context->GetIsolate(),
+                                                  source_context));
+      proxy.SetMethod(
+          key_str, base::BindRepeating(&ProxyFunctionWrapper, store, func_id));
+    } else if (value->IsObject() && !value->IsNullOrUndefined() &&
+               !value->IsArray() && !value->IsPromise()) {
+      mate::Dictionary sub_api(api.isolate(),
+                               v8::Local<v8::Object>::Cast(value));
+      proxy.Set(key_str, CreateProxyForAPI(sub_api, source_context,
+                                           target_context, store));
+    } else {
+      proxy.Set(key_str, PassValueToOtherContext(source_context, target_context,
+                                                 value, store));
+    }
+  }
+
+  return proxy;
+}
+
+#ifdef DCHECK_IS_ON
+mate::Dictionary DebugGC(mate::Dictionary empty) {
+  auto* render_frame = GetRenderFrame(empty.GetHandle());
+  RenderFramePersistenceStore* store = GetOrCreateStore(render_frame);
+  mate::Dictionary ret = mate::Dictionary::CreateEmpty(empty.isolate());
+  ret.Set("functionCount", store->functions().size());
+  return ret;
+}
+#endif
+
+void ExposeAPIInMainWorld(const std::string& key,
+                          mate::Dictionary api,
+                          mate::Arguments* args) {
+  auto* render_frame = GetRenderFrame(api.GetHandle());
+  RenderFramePersistenceStore* store = GetOrCreateStore(render_frame);
+  auto* frame = render_frame->GetWebFrame();
+  DCHECK(frame);
+  v8::Local<v8::Context> main_context = frame->MainWorldScriptContext();
+  mate::Dictionary global(main_context->GetIsolate(), main_context->Global());
+
+  if (global.Has(key)) {
+    args->ThrowError(
+        "Cannot bind an API on top of an existing property on the window "
+        "object");
+    return;
+  }
+
+  v8::Local<v8::Context> isolated_context =
+      frame->WorldScriptContext(api.isolate(), World::ISOLATED_WORLD);
+
+  {
+    v8::Context::Scope main_context_scope(main_context);
+    mate::Dictionary proxy =
+        CreateProxyForAPI(api, isolated_context, main_context, store);
+    DeepFreeze(proxy.GetHandle(), main_context);
+    global.SetReadOnlyNonConfigurable(key, proxy);
+  }
+}
+
+}  // namespace api
+
+}  // namespace electron
+
+namespace {
+
+void Initialize(v8::Local<v8::Object> exports,
+                v8::Local<v8::Value> unused,
+                v8::Local<v8::Context> context,
+                void* priv) {
+  using namespace electron::api;  // NOLINT(build/namespaces)
```

Comment:
instead of disabling lint, just qualify the function name

Reply:
One of the last remnants of a copy-paste from somewhere else